src/net/fe/unit/Unit.java

```diff
@@ -286,6 +286,8 @@ public void drop(int x, int y){
 		rescuedUnit.rescued = false;
 		rescuedUnit.setMoved(true);
 		final OverworldStage grid = (OverworldStage) stage;
+		if(!grid.getCurrentPlayer().getParty().getUnits().contains(this))
+			rescuedUnit.setMoved(false);
```

Comment:
Lines 287, 289 and 290 are equivalent to `rescuedUnit.setMoved(! grid.getCurrentPlayer().getParty().getUnits().contains(this));`, right? And the version without the conditional `if` is more straightforward, in my opinion.

Reply:
That is actually incorrect. ``setMoved(true)`` actually performs action to return the unit to an idle state. Because of that, both calls to the method are required. An alternative would be to call ``sprite.setAnimation("IDLE")`` (I don't think the ``origX/Y = x/ycoord`` statements are required in this situation, but I could be wrong).

Reply:
You are correct. yes.